lib/puppet/pops/types/types.rb

```diff
@@ -1307,8 +1524,9 @@ class PVariantType < PAnyType
 
       attr_reader :types
 
+      # @param types [Array[PAnyType]] the variants
       def initialize(types)
-        @types = types.freeze
+        @types = types.uniq.freeze
```

Comment:
Assume you do infer an element type for [0,1,1,1,1,1,1,2,2,2,2,2,2]. Do you really want that to result in a Variant with 13 Integer types? Or is 3 enough?